pkg/controller/daemon/daemoncontroller.go

```diff
@@ -650,6 +652,22 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false
 	}
 
+	newPod := &v1.Pod{Spec: ds.Spec.Template.Spec, ObjectMeta: ds.Spec.Template.ObjectMeta}
```

Comment:
Make this an utility that accepts a daemonSet and a node name and it returns a Pod?